lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -946,21 +950,33 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
                 -- in the final stage of 'balanceTransaction'.
                 extraBytes = 8
             in
-            W.Coin $ round perByte * (extraBytes + scriptIntegrityHashBytes)
+                W.toWallet . feeOfBytes feePerByte $
+                    extraBytes + scriptIntegrityHashBytes
 
         fromCardanoLovelace (Cardano.Lovelace l) = Coin.unsafeFromIntegral l
 
         selectionConstraints = SelectionConstraints
             { assessTokenBundleSize =
-                view #assessTokenBundleSize $
-                tokenBundleSizeAssessor txLayer $
-                pp ^. #txParameters . #getTokenBundleMaxSize
-            , certificateDepositAmount =
-                pp ^. #stakeKeyDeposit
-            , computeMinimumAdaQuantity =
-                constraints txLayer era pp ^. #txOutputMinimumAdaQuantity
-            , isBelowMinimumAdaQuantity =
-                constraints txLayer era pp ^. #txOutputBelowMinimumAdaQuantity
+                view #assessTokenBundleSize
+                $ tokenBundleSizeAssessor txLayer
+                $ TokenBundleMaxSize
+                $ TxSize
+                $ case recentEra @era of
+                    RecentEraBabbage -> pp ^. #_maxValSize
+                    RecentEraConway -> pp ^. #_maxValSize
+            , certificateDepositAmount = W.toWallet $ case recentEra @era of
+                RecentEraBabbage -> getField @"_keyDeposit" pp
+                RecentEraConway -> getField @"_keyDeposit" pp
+            , computeMinimumAdaQuantity = \addr tokens -> W.toWallet $
+                computeMinimumCoinForTxOut
+                    (recentEra @era)
+                    pp
+                    (toLedgerTxOut $ W.TxOut addr (TokenBundle txOutMaxCoin tokens))
```

Comment:
This `txOutMaxCoin` here will be ignored by the implementation of `computeMinimumCoinForTxOut`, so it doesn't matter. In practice the implementation will itself update the value to `txOutMaxCoin` regardless of what it was previously.